src/subcommands/logs_test.ts

```diff
@@ -1,13 +1,33 @@
 import { parseArgsForLogSubcommand } from "./logs.ts";
-import { assertEquals, assertThrows } from "jsr:@std/assert@0.217";
+import {
+  assertEquals,
+  assertNotEquals,
+  assertThrows,
+} from "jsr:@std/assert@0.217";
 import { parseArgs } from "../args.ts";
 
 Deno.test("parseArgsForLogSubcommand", async (t) => {
-  const parseHelper = (args: string[]) =>
-    // NOTE: We omit `logs` subcommand from the arguments passed to `parseArgs()`
-    // in order to match the actual behavior; the first positional argument is
-    // removed using `args._.shift()` in `deployctl.ts`.
-    parseArgsForLogSubcommand(parseArgs(args));
+  const parseHelper = (args: string[]) => {
+    // For this test, the subcommand name should not be included in `args`.
+    assertNotEquals(args.at(0), "logs");
+
+    try {
+      return parseArgsForLogSubcommand(parseArgs(args));
+    } catch (e) {
+      // Since Deno v1.44.0, when `Deno.exitCode` was introduced, test cases
+      // with non-zero exit code has been treated as failure, causing some tests
+      // to fail unexpectedly (not sure if this behavior change is intended).
+      // To avoid this, we set `Deno.exitCode` to 0 before giving control back
+      // to each test case.
+      // https://github.com/denoland/deno/pull/23609
+      // deno-lint-ignore no-explicit-any
+      if ((Deno as any).exitCode !== undefined) {
+        // deno-lint-ignore no-explicit-any
+        (Deno as any).exitCode = 0;
```

Comment:
Where is it set to 1? We can avoid setting the code to 1

Reply:
This test calls to `parseArgsForLogSubcommand`, which bails out when it encounters any invalid arguments. For instance at the line 184 it calls `error` which internally calls `Deno.exit(1)`.

https://github.com/denoland/deployctl/blob/0c4210bf4f92fa9f75d41e644623dfb976bef2ae/src/subcommands/logs.ts#L175-L186

We could refactor to avoid using `Deno.exit` in `parseArgsForLogSubcommand` but instead bubble up the error into the caller, but given that this is a CLI application it's just easier to bail out whenever it find some error. So I would prefer to keep `parseArgsForLogSubcommand` as is.